src/Common/src/TypeSystem/IL/Stubs/ILEmitter.cs

```diff
@@ -31,10 +31,13 @@ public LabelAndOffset(ILCodeLabel label, int offset)
 
         private ArrayBuilder<LabelAndOffset> _offsetsNeedingPatching;
 
-        internal ILCodeStream()
+        private ILEmitter _emitter;
```

Comment:
Having more high level APIs on ILCodeStream sounds good to me. It is similar to how IL stubs are done in CoreCLR.